app/src/main/java/uk/ac/ebi/atlas/configuration/WebConfig.java

```diff
@@ -69,6 +74,50 @@ public UrlBasedViewResolver urlBasedViewResolver() {
         return resolver;
     }
 
+    @Bean
+    public ClassLoaderTemplateResolver rootThymeleafTemplateResolver() {
+        ClassLoaderTemplateResolver resolver = new ClassLoaderTemplateResolver();
+        resolver.setPrefix("templates/thymeleaf/");
+        resolver.setSuffix(".html");
+        resolver.setTemplateMode(TemplateMode.HTML);
+        resolver.setCharacterEncoding("UTF-8");
+        resolver.setCheckExistence(true);
+        return resolver;
+    }
+
+    @Bean
+    public ClassLoaderTemplateResolver viewsThymeleafTemplateResolver() {
+        ClassLoaderTemplateResolver resolver = new ClassLoaderTemplateResolver();
+        resolver.setPrefix("templates/thymeleaf/views/");
+        resolver.setSuffix(".html");
+        resolver.setTemplateMode(TemplateMode.HTML);
+        resolver.setCharacterEncoding("UTF-8");
+        resolver.setCheckExistence(true);
+        return resolver;
+    }
+
+    @Bean
+    public SpringTemplateEngine thymeleafTemplateEngine() {
+        SpringTemplateEngine templateEngine = new SpringTemplateEngine();
+        templateEngine.addTemplateResolver(viewsThymeleafTemplateResolver());
+        templateEngine.addTemplateResolver(rootThymeleafTemplateResolver());
+        return templateEngine;
+    }
```

Comment:
I thinkg that is a problem here. `thymeleafTemplateEngine` will not use the 2 `ClassLoaderTemplateResolver` beans created above, but some unmanaged instances.
If `viewsThymeleafTemplateResolver` is a bean, you can should pass it as an argument to this function. In this case since both arguments have the same type, you would need to use a `@Qualifier` to distinguish between them.
```suggestion

    @Bean
    public SpringTemplateEngine thymeleafTemplateEngine(
        @Qualifier("viewsThymeleafTemplateResolver") ClassLoaderTemplateResolver viewsThymeleafTemplateResolver,
        @Qualifier("rootThymeleafTemplateResolver") ClassLoaderTemplateResolver rootThymeleafTemplateResolver
    ) {
        SpringTemplateEngine templateEngine = new SpringTemplateEngine();
        templateEngine.addTemplateResolver(viewsThymeleafTemplateResolver);
        templateEngine.addTemplateResolver(rootThymeleafTemplateResolver);
        return templateEngine;
    }
```

Reply:
I don't agree with you.
Please look at examples: 
- https://www.baeldung.com/thymeleaf-in-spring-mvc
- https://www.thymeleaf.org/doc/articles/springmail.html
They are setting up the configuration in the same way as me and one of them is the official doc for ThymeLeaf.

Reply:
@amnonkhen 
Please take a look at the Spring documentation: https://docs.spring.io/spring-framework/reference/core/beans/java/configuration-annotation.html
Thanks

Reply:
Although it will work (due to spring's magic. use of proxying of function calls) the approach in the baeldung tutorial is not as good as what I suggested, which :
1. It makes dependencies explicit in the function's signature rather than an obscure method call in the function's body.
2. It improves testability and reduces confusion, becuase you can mock each dependency.
3. It avoids relying on Spring’s proxy behaviour (less important IMO)

Reply:
@amnonkhen I still not agree with your solution and any of your above points. Your proposed solution is kind of ugly and not necessary.
Please check Spring's documentation that I also referred above.
I copy it here again: https://docs.spring.io/spring-framework/reference/core/beans/java/configuration-annotation.html
Thanks